aptible/resource_database.go

```diff
@@ -88,68 +94,140 @@ func resourceDatabase() *schema.Resource {
 }
 
 func resourceDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*providerMetadata).LegacyClient
+	client := meta.(*providerMetadata).Client
+	legacy := meta.(*providerMetadata).LegacyClient
 	envID := int64(d.Get("env_id").(int))
 	handle := d.Get("handle").(string)
 	version := d.Get("version").(string)
 	databaseType := d.Get("database_type").(string)
+	iops := int32(d.Get("iops").(int))
+	profile := d.Get("container_profile").(string)
+	diskSize := int32(d.Get("disk_size").(int))
+	containerSize := int32(d.Get("container_size").(int))
+
+	ctx := context.Background()
+	ctx = meta.(*providerMetadata).APIContext(ctx)
 
-	attrs := aptible.DBCreateAttrs{
-		Handle:           &handle,
-		Type:             databaseType,
-		ContainerSize:    int64(d.Get("container_size").(int)),
-		ContainerProfile: d.Get("container_profile").(string),
-		DiskSize:         int64(d.Get("disk_size").(int)),
+	create := aptibleapi.NewCreateDatabaseRequestWithDefaults()
+	create.SetHandle(handle)
+	createDb := client.DatabasesAPI.CreateDatabase(ctx, int32(envID))
+	create.SetType(databaseType)
+
+	if diskSize != 0 {
```

Comment:
FYI: I believe you can also use `d.HasChange("disk_size")`.

Reply:
does that work even for model creation?

Reply:
Yeah, I played around with it. For creation, it defaults everything to `null` so if nothing is provided `HasChange` is false. It doesn't matter here since a disk size of 0 is invalid. Nothing needs to change, just an FYI about how the provider SDK works.